tests/test_optimize.py

```diff
@@ -462,6 +468,113 @@ def test_adam_optimizer_multivar(self, bunch, tol):
                 x_twosteps_target = x_onestep - adapted_stepsize * firstmoment / (np.sqrt(secondmoment) + 1e-8)
                 assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
 
+    @pytest.mark.parametrize('x_start', x_vals)
+    def test_rotosolve_optimizer_univar(self, x_start, bunch, tol):
+        """Tests that rotosolve optimizer takes one and two steps correctly
+        for uni-variate functions."""
+
+        for f in univariate_funcs:
+            x_onestep = bunch.rotosolve_opt.step(f, x_start)
+            x_onestep_target = self.rotosolve_step(f, x_start)
+
+            assert x_onestep == pytest.approx(x_onestep_target, abs=tol)
+
+            x_twosteps = bunch.rotosolve_opt.step(f, x_onestep)
+            x_twosteps_target = self.rotosolve_step(f, x_onestep_target)
+
+            assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
+
+    def test_rotosolve_optimizer_multivar(self, bunch, tol):
+        """Tests that rotosolve optimizer takes one and two steps correctly
+        for multi-variate functions."""
+
+        for f in multivariate_funcs:
+            for jdx in range(len(x_vals[:-1])):
+                x_vec = x_vals[jdx:jdx + 2]
+
+                x_onestep = bunch.rotosolve_opt.step(f, x_vec)
+                x_onestep_target = self.rotosolve_step(f, x_vec)
+
+                assert x_onestep == pytest.approx(x_onestep_target, abs=tol)
+
+                x_twosteps = bunch.rotosolve_opt.step(f, x_onestep)
+                x_twosteps_target = self.rotosolve_step(f, x_onestep_target)
+
+                assert x_twosteps == pytest.approx(x_twosteps_target, abs=tol)
+
+    @pytest.mark.parametrize('x_start', [0.5, -1.14, [0.3, 0.25], [-0.6, 0.45], [1.3, -0.9]])
+    @pytest.mark.parametrize('generators', [list(tup) for tup in it.product([qml.RX, qml.RY, qml.RZ], repeat=2)])
+    def test_rotoselect_optimizer(self, x_start, generators, bunch, tol):
+        """Tests that rotoselect optimizer takes one and two steps correctly for the VQE circuit
+        defined in `this rotoselect tutorial <https://pennylane.ai/qml/demos/tutorial_rotoselect.html>`_."""
+
+        # the optimal generators for the 2-qubit VQE circuit
+        # H = 0.5 * Y_2 + 0.8 * Z_1 - 0.2 * X_1
+        optimal_generators = [qml.RY, qml.RX]
+        possible_generators = [qml.RX, qml.RY, qml.RZ]
+        bunch.rotoselect_opt.possible_generators = possible_generators
+
+        dev = qml.device("default.qubit", analytic=True, wires=2)
+
+        def ansatz_rsel(params, generators):
```

Comment:
Minor suggestion: the `rsel` suffix in the function names might not indicate too much related to its content. Therefore it could be removed or replaced with another suffix. In terms of the two circuits, simply numbering them could also work such as `circuit_1`.